packages/client/src/sigstore-utils.ts

```diff
@@ -30,45 +39,50 @@ export async function createDSSEEnvelope(
     signer: SignerFunc;
   }
 ): Promise<SerializedEnvelope> {
-  // Pre-authentication encoding to be signed
-  const paeBuffer = dsse.preAuthEncoding(payloadType, payload);
-
-  // Get signature and verification material for pae
-  const sigMaterial = await options.signer(paeBuffer);
-
-  const envelope: sigstore.Envelope = {
-    payloadType,
-    payload,
-    signatures: [
-      {
-        keyid: sigMaterial.key?.id || '',
-        sig: sigMaterial.signature,
-      },
-    ],
-  };
-
-  return sigstore.Envelope.toJSON(envelope) as SerializedEnvelope;
+  const bundler = createBundleBuilder('dsseEnvelope', {
+    signer: options.signer,
+    tlogUpload: false,
+  });
+  const bundle = await bundler.create({ data: payload, type: payloadType });
+  return envelopeToJSON(bundle.content.dsseEnvelope);
 }
 
 // Accepts a signed DSSE envelope and a PEM-encoded public key to be added to the
 // transparency log. Returns a Sigstore bundle suitable for offline verification.
 export async function createRekorEntry(
   dsseEnvelope: SerializedEnvelope,
   publicKey: string,
+  /* istanbul ignore next */
   options: SignOptions = {}
 ): Promise<SerializedBundle> {
-  const envelope = sigstore.Envelope.fromJSON(dsseEnvelope);
-  const tlog = createTLogClient(options);
+  const envelope = envelopeFromJSON(dsseEnvelope);
+  const content: SignatureBundle = {
+    $case: 'dsseEnvelope',
+    dsseEnvelope: envelope,
+  };
 
-  const sigMaterial = extractSignatureMaterial(envelope, publicKey);
-  const entry = await tlog.createDSSEEntry(envelope, sigMaterial, {
+  const tlog = new RekorWitness({
+    rekorBaseURL:
+      options.rekorURL || /* istanbul ignore next */ DEFAULT_REKOR_URL,
     fetchOnConflict: true,
+    retry: options.retry ?? DEFAULT_RETRY,
+    timeout: options.timeout ?? DEFAULT_TIMEOUT,
   });
 
-  const bundle = sigstore.toDSSEBundle({
-    envelope,
-    signature: sigMaterial,
-    tlogEntry: entry,
-  });
+  const vm = await tlog.testify(content, publicKey);
+
+  const bundle: Bundle = {
+    mediaType: BUNDLE_V01_MEDIA_TYPE,
+    content,
```

Comment:
I think this will result in something like this:

```
mediaType: ...
content: {
  $case: 'dsseEnvelope',
  dsseEnvelope: envelope,
}
``` 

Do we also need to omit the `$case` key or is this handled by `bundleToJSON`:
```suggestion
    dsseEnvelope: content.dsseEnvelope,
```

Seems worth adding some tests around this to match against the expected bundle output.

Reply:
or wait, is this all handled by `bundleToJSON`?

Reply:
I think I'm just getting tripped up by the manual bundle construction here. Would it be worth adding some helpers to the `bundle` package? Wondering later down the line if it would be easier to have everything bundle related in one place so we can do roll out bigger changes and not miss things elsewhere.